docs/management_api.yml

```diff
@@ -28,83 +28,80 @@ tags:
   - name: ManagementAPI
 
 paths:
-  /configuration:
+  /config/devices/{deviceId}:
     get:
-      operationId: List Parameters
+      operationId: Get key-value pair store
       tags:
         - ManagementAPI
-      summary: Get existing parameters
-      # parameters: {}
+      summary: Query the configuration store; retrieve all key-value pairs
+      parameters:
+        - in: path
+          name: deviceId
+          schema:
+            type: string
+          required: true
+          description: ID of the device to query.
       responses:
         200:
           description: OK
           content:
             application/json:
               schema:
-                type: array
-                items:
-                  $ref: '#/components/schemas/Parameter'
+                type: object
+                properties:
+                  id:
+                    type: string
+                  expected:
+                    type: array
+                    items:
+                      $ref: '#/components/schemas/Parameter'
+                  actual:
+                    type: array
+                    items:
+                      $ref: '#/components/schemas/Parameter'
+                  updated_ts:
+                    type: string
+                  report_ts:
+                    type: string
         400:
-          description: Invalid Request.
+          description: Bad Request.
           content:
             application/json:
               schema:
                 $ref: '#/components/schemas/Error'
-        500:
-          description: Internal Server Error.
+        404:
+          description: Not Found.
           content:
             application/json:
               schema:
                 $ref: '#/components/schemas/Error'
-
     put:
-      operationId: Set Configuration
+      operationId: Store a set of key-value pairs
       tags:
         - ManagementAPI
-      summary: Assign configuration parameters
+      summary: Set a key-value pair store, updating if existing, removing if empty
       requestBody:
         content:
           application/json:
             schema:
-              oneOf:
-                - $ref: '#/components/schemas/Parameter'
-                - type: array
+              type: object
+              properties:
+                expected:
+                  type: array
```

Comment:
See my comment above: the interface should allow us to specify a JSON in the form `{"key": "value"}`, not a list of attributes.

Reply:
that was original implementation and original conclusion with Manuel. but the existing code does not go in that direction. I will fix.

Reply:
fixed.

Reply:
As mentioned above, fixed in the code, but the API docs is outdated.